sdlib/d/gc/region.d

```diff
@@ -519,22 +566,21 @@ unittest rangeTree {
 }
 
 ptrdiff_t classAddrRegionCmp(Region* lhs, Region* rhs) {
-	static assert(LgAddressSpace <= 56, "Address space too large!");
-
-	auto l = ulong(lhs.allocClass) << 56;
-	auto r = ulong(rhs.allocClass) << 56;
+	enum ClassAddrMask = ulong(Region.AllocClassMask) << Region.AllocClassIndex
+		| Region.SignificantAddressMask;
 
-	l |= cast(size_t) lhs.address;
-	r |= cast(size_t) rhs.address;
+	auto l = lhs.bits & ClassAddrMask;
+	auto r = rhs.bits & ClassAddrMask;
```

Comment:
You need to mask because you put the generation in the high bits, but you didn't need to.